src/main/java/htsjdk/samtools/fastq/FastqEncoder.java

```diff
@@ -91,13 +92,25 @@ public static FastqRecord asFastqRecord(final SAMRecord record) {
         if(record.getReadPairedFlag() && (record.getFirstOfPairFlag() || record.getSecondOfPairFlag())) {
             readName += (record.getFirstOfPairFlag()) ? FastqConstants.FIRST_OF_PAIR : FastqConstants.SECOND_OF_PAIR;
         }
-        return new FastqRecord(readName, record.getReadString(), null, record.getBaseQualityString());
+        return new FastqRecord(readName, record.getReadString(), record.getStringAttribute(SAMTag.CO.name()), record.getBaseQualityString());
```

Comment:
Are you concerned that a record that has a comment that it not related to the QualityHeader will be misunderstood and the comment will then become the quality header? 

The wikipedia page seems to imply that the sample_id needs to be the "first" part of this header, followed by arbitrary text, but on the other hand, the fastq format is poorly defined...so it's not even clear what the separator is.

Reply:
I just would like to have a lossless conversion from FASTQ to SAM with this encoder. Validation of the comment could be performed downstream.